community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyStore.java

```diff
@@ -301,7 +301,7 @@ public void ensureHeavy( PropertyBlock block )
                     while ( stringRecords.next() )
                     {
                         stringRecords.get().setType( PropertyType.STRING.intValue() );
-                        block.addValueRecord( stringRecords.get() );
+                        block.addValueRecord( stringRecords.get().clone() );
```

Comment:
stringRecords is a cursor, so returned DynamicRecord is reused for each iteration.